plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt

```diff
@@ -117,15 +117,19 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
             currentCoroutineContext.ensureActive()
             val artifactsUploadStartTime = now()
             val codeScanName = UUID.randomUUID().toString()
-            val sourceZipUploadResponse = createUploadUrlAndUpload(sourceZip, "SourceCode", codeScanName)
-            if (isProjectScope()) {
-                LOG.debug {
-                    "Successfully uploaded source zip to s3: " +
-                        "Upload id: ${sourceZipUploadResponse.uploadId()} " +
-                        "Request id: ${sourceZipUploadResponse.responseMetadata().requestId()}"
+            try {
+                val sourceZipUploadResponse = createUploadUrlAndUpload(sourceZip, "SourceCode", codeScanName)
+                if (isProjectScope()) {
+                    LOG.debug {
+                        "Successfully uploaded source zip to s3: " +
+                            "Upload id: ${sourceZipUploadResponse.uploadId()} " +
+                            "Request id: ${sourceZipUploadResponse.responseMetadata().requestId()}"
+                    }
                 }
+                urlResponse[ArtifactType.SOURCE_CODE] = sourceZipUploadResponse
+            } finally {
+                sourceZip.delete()
```

Comment:
please clarify what type of "memory use" you are referring to.

if this is related to the complaints of high RAM usage, how does removing a file from disk help?